src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -658,7 +658,9 @@ public virtual bool IsAsyncBreakSignal(Results results)
 
             if (results.TryFindString("reason") == "signal-received")
             {
-                if (results.TryFindString("signal-name") == "SIGTRAP")
+                if (results.TryFindString("signal-name") == "SIGTRAP" || 
+                    // If we are remote, we use -exec-interrupt to async-break. GDB will send a SIGINT and we will need to handle the async break
+                    (_debugger.IsRemoteGdbTarget() && (_debugger.IsRequestingInternalAsyncBreak || _debugger.IsRequestingRealAsyncBreak) && results.TryFindString("signal-name") == "SIGINT"))
```

Comment:
>_debugger.IsRemoteGdbTarget() && (_debugger.IsRequestingInternalAsyncBreak || _debugger.IsRequestingRealAsyncBreak) && [](http://example.com/codeflow?start=21&length=118)

Should we instead set before we do `CmdAsync("-exec-interrupt", ResultClass.done)` since the logic for when we use `-exec-interrupt` is pretty complicated (ex: looks like it is used for LLDB, and only only sometimes for remote GDB).

Reply:
Would we clear that bool flag here?

Reply:
I think it probably makes more sense to do in the caller. Maybe be returning an enum instead of a bool?

Reply:
Yeah, I think we would need to since it looks like `IsAsyncBreakSignal` could be called twice for the same stop.

Reply:
I created the `AsyncBreakSignal` enum to handle this.